helix-view/src/editor.rs

```diff
@@ -391,7 +393,21 @@ impl Editor {
         self.status_msg = Some((error.into(), Severity::Error));
     }
 
-    pub fn set_theme(&mut self, theme: Theme) {
+    pub fn unset_theme_preview(&mut self) {
+        let theme = self.theme_loader.load(&self.current_theme).unwrap();
+        self.set_theme(theme, self.current_theme.clone());
+    }
+
+    pub fn set_theme_preview(&mut self, theme: Theme) {
+        self.set_theme_impl(theme);
+    }
+
+    pub fn set_theme(&mut self, theme: Theme, theme_name: String) {
+        self.current_theme = theme_name;
+        self.set_theme_impl(theme);
+    }
+
+    fn set_theme_impl(&mut self, theme: Theme) {
```

Comment:
I've refactored this a bit but I can't meet this exact behaviour. I was able to get it so we don't have to pass a theme name around anymore, but the underlying implementation still has to work like it did before. I still have a `default_theme` function, since the theme is determined by some application-specific parameters and isn't quite "`impl Default for Theme`", but it doesn't deal with the theme names anymore.